### PR TITLE
Fix using and, or, xor in desul with MSVC

### PR DESCRIPTION
**Pull request by masterleinad**
Corresponds to https://github.com/desul/desul/pull/130.
https://github.com/kokkos/kokkos/actions/runs/9845036970/job/27179696028?pr=7123 and https://godbolt.org/z/5384bWqWY show that using `and`, `or`, and `xor` as macro arguments is problematic with MSVC when `<ciso646>` is included. This pull request makes sure that we post-fix and pre-fix these arguments instead. ~~We might need to replace more if the new arguments are reserved in C++.~~

### CODE REVIEW

**Comment by crtrott** [comment]
lol shit

File: tpls/desul/include/desul/atomics/Fetch_Op_HIP.hpp [hunk 1]
```diff
@@ -14,36 +14,36 @@ SPDX-License-Identifier: (BSD-3-Clause)
...
 
-DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(add)
-DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(min)
-DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(max)
-DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(and)
+DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(_add)
+DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(_min)
+DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(_max)
+DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(_and)
 DESUL_IMPL_HIP_ATOMIC_FETCH_OP_INTEGRAL(or)
```
Comment (ldh4) [inline tpls/desul/include/desul/atomics/Fetch_Op_HIP.hpp:41]: Looks like you missed a line?

Comment (masterleinad) [reply tpls/desul/include/desul/atomics/Fetch_Op_HIP.hpp]: fixed

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
The `desul` pull request has been merged.

**Review by dalg24 — APPROVED** [review]

**Review by ldh4 — APPROVED** [review]

**Pull request merged**